test/opt/local_single_block_elim.cpp

```diff
@@ -183,87 +183,176 @@ OpFunctionEnd
       predefs + before, predefs + after, true, true);
 }
 
+TEST_F(LocalSingleBlockLoadStoreElimTest, StoreStoreElim) {
+  //
+  // Note first store to v is eliminated
+  //
+  // #version 450
+  //
+  // layout(location = 0) in vec4 BaseColor;
+  // layout(location = 0) out vec4 OutColor;
+  //
+  // void main()
+  // {
+  //     vec4 v = BaseColor;
+  //     v = v * 0.5;
+  //     OutColor = v;
+  // }
```

Comment:
If we convert to SSA and run DCE, this should eliminate the fist store to ```v``` does it not?

Reply:
It does, actually.  Running ```spirv-opt --ssa-rewrite --eliminate-dead-code-aggressive``` on the test case results in:

```
       %main = OpFunction %void None %7
         %14 = OpLabel
         %15 = OpLoad %v4float %BaseColor
         %17 = OpVectorTimesScalar %v4float %15 %float_0_5
               OpStore %OutColor %17
               OpReturn
               OpFunctionEnd
```

Reply:
Yes, SSARewrite functionally can replace SingleBlockElim and SingleStoreElim, but that is not the way the current recipes are written.

Several months ago this change was made to the recipes and I had to go back and demonstrate that running SingleBlock and SingleStore before SSARewrite reduced overall compile time. This is also the way that LLVM does it, at least the last time I looked.

So I would request that if you decide to try this again you look at overall compile time. Until that time I am requesting this very simple and inexpensive improvement to SingleBlock.

Reply:
Oh, interesting.  If this reduces overall compile-time, then we shouldn't even consider getting rid of SBE and SSE.  I'll merge your change.  Thanks!